integration-tests/velocity/src/main/java/org/apache/camel/quarkus/component/velocity/it/VelocityResource.java

```diff
@@ -112,6 +115,34 @@ public Response bodyAsDomainObject(Person person, @QueryParam("givenName") Strin
                 .build();
     }
 
+    @Path("/velocityContext")
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response velocityContext(String msg, @QueryParam("name") String name, @QueryParam("name2") String name2,
+            @QueryParam("item") String item) throws Exception {
+
+        final Exchange ex = producerTemplate.request(
+                "velocity://template/velocityContext.vm?allowTemplateFromHeader=true&allowContextMapAll=true",
+                (Processor) exchange -> {
+                    exchange.getIn().setBody("");
+                    exchange.getIn().setHeader("name", name2);
+                    Map<String, Object> variableMap = new HashMap<>();
+                    variableMap.put("headers", CollectionHelper.mapOf("name", name));
+                    variableMap.put("body", "Monday");
+                    variableMap.put("properties", exchange.getProperties());
```

Comment:
(Just pointing to the place, where exchange was originally put into context, I switched it for the properties map, therefore now there is no problem)